src/Firebase/Streaming/FirebaseSingleObjectCache.cs

```diff
@@ -0,0 +1,87 @@
+﻿namespace Firebase.Database.Streaming
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Newtonsoft.Json;
+
+    public class FirebaseSingleObjectCache<T> : FirebaseCache<T>
```

Comment:
Instead of inheriting from FirebaseCache, can you introduce `IFirebaseCache<T>` which both classes would implement?